apps/docs/content/guides/auth/jwts.mdx

```diff
@@ -1,178 +1,236 @@
+In the past there was a recommendation to set custom headers on the Supabase client with the `Authorization` header including your custom JWT. This is no longer recommended as it's less flexible and causes confusion when combined with a user session from Supabase Auth.
 
-In Supabase we issue JWTs for three different purposes:
 
-1. `anon key`: This key is used to bypass the Supabase API gateway and can be used in your client-side code.
-2. `service role key`: This key has super admin rights and can bypass your Row Level Security. Do not put it in your client-side code. Keep it private.
-3. `user specific jwts`: These are tokens we issue to users who log into your project/service/website. It's the modern equivalent of a session token, and can be used by a user to access content or permissions specific to them.
+## Verifying a JWT from Supabase
 
-The first token here, the `anon key` token, is for developers to send along with their API requests whenever they want to interact with their Supabase database.
+If you're not able to use the Supabase client libraries, the following can be used to help you securely verify JWTs issued by Supabase.
 
-Let's say you want to read the names of all the rows in a table `colors`. We would make a request like:
+Supabase Auth exposes a [JSON Web Key](https://datatracker.ietf.org/doc/html/rfc7517) Set URL for each Supabase project:
 
-```bash
-curl 'https://xscduanzzfseqszwzhcy.supabase.co/rest/v1/colors?select=name' \
--H "apikey: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYW5vbiIsImlhdCI6MTYxNDIwNTE3NCwiZXhwIjoxOTI5NzgxMTc0fQ.-NBR1WnZyQGpRLdXJfgfpszoZ0EeE6KHatJsDPLIX8c"
+```http
+GET https://project-id.supabase.co/auth/v1/.well-known/jwks.json
 ```
 
-If we put this token into https://jwt.io, we see it decodes to:
+Which responds with JWKS object containing one or more asymmetric [JWT signing keys](/docs/guides/auth/signing-keys) (only their public keys).
 
-```js
+```json
 {
-  "role": "anon",
-  "iss": "supabase",
-  "iat": 1614205174,
-  "exp": 1929781174
+  "keys": [
+    {
+      "kid": "<match with kid from JWT header>",
+      "alg": "<match with alg from JWT header>",
+      "kty": "<RSA|EC|OKP>",
+      "key_ops": ["verify"],
+      // public key fields
+    }
+  ]
 }
 ```
 
-This JWT is signed by a `jwt_secret` specific to the developer's Supabase token (you can find this secret alongside this encoded "anon key" on your Dashboard under Settings > API page) and is required to get past the Supabase API gateway and access the developer's project.
+This endpoint is served directly from the Auth server, but is also additionally cached by the Supabase Edge for 10 minutes, significanlty speeding up access to this data regardless of where you're performing the verification. It's important to be aware of the cache expiry time to prevent unintentionally rejecting valid user access tokens. We recommend waiting at least 20 minutes when creating a standby signing key, or revoking a previously used key.
```

Comment:
```suggestion
This endpoint is served directly from the Auth server, but is also additionally cached by the Supabase Edge for 10 minutes, significantly speeding up access to this data regardless of where you're performing the verification. It's important to be aware of the cache expiry time to prevent unintentionally rejecting valid user access tokens. We recommend waiting at least 20 minutes when creating a standby signing key, or revoking a previously used key.
```